pymeasure/instruments/agilent/agilent4284A.py

```diff
@@ -0,0 +1,288 @@
+        """Control the impedance measurement range. The 4284A will select an appropriate
+        measurement range for the setting value."""
+    )
+
+    auto_range_enabled = Instrument.control(
+        "FUNC:IMP:RANG:AUTO?", "FUNC:IMP:RANG:AUTO %d",
+        """Control whether the impedance auto range is enabled.""",
+        validator=strict_discrete_set,
+        values={False: 0, True: 1}
+    )
+
+    def _set_ranges(self, condition):
+        """Set dynamic property values and make copies for sweep_measurement to reference."""
+        if condition == 0:
+            self.ac_current_values = (50e-6, 0.02)
+            self.ac_voltage_values = (0.005, 2)
+            self.bias_voltage_values = (0, 2)
+            self.bias_current_values = (0, 0)
+            self._ac_current_values = (50e-6, 0.02)
+            self._ac_voltage_values = (0.005, 2)
+            self._bias_voltage_values = (0, 2)
+            self._bias_current_values = (0, 0)
+        elif condition == 1:
+            self.ac_current_values = (50e-6, 0.2)
+            self.ac_voltage_values = (0.005, 20)
+            self.bias_voltage_values = (0, 40)
+            self.bias_current_values = (0, 0.1)
+            self._ac_current_values = (50e-6, 0.2)
+            self._ac_voltage_values = (0.005, 20)
+            self._bias_voltage_values = (0, 40)
+            self._bias_current_values = (0, 0.1)
+
+    def enable_high_power(self):
+        """Enable high power mode.
+
+        Requires option 001 (power amplifier / DC bias) is installed.
+        """
+        if self.options[0] == '0':
+            log.warning("Agilent 4284A power amplifier is not installed.")
+            return
+        self._set_ranges(1)
+        self.write("OUTP:HPOW 1")
+
+    def disable_high_power(self):
+        """Disable high power mode."""
+        self._set_ranges(0)
+        self.write("OUTP:HPOW 0")
+
+    def sweep_measurement(self, sweep_mode, sweep_values):
+        """Run list sweep measurement using sequential trigger.
+
+        :param sweep_mode (str): parameter to sweep across. Must be one of `frequency`,
+            `voltage`, `current`, `bias_voltage`, or `bias_current`.
+        :param sweep_values: list of parameter values to sweep across.
+
+        :returns: values as configured with :attr:`~.Agilent4284A.impedance_mode` and
+            list of sweep parameters in format ([val A], [val B], [sweep_values])
+        """
+        param_dict = {
+            "frequency": ("FREQ", (20, 1e6)),
+            "voltage": ("VOLT", self._ac_voltage_values),
+            "current": ("CURR", self._ac_current_values),
+            "bias_voltage": ("BIAS:VOLT", self._bias_voltage_values),
+            "bias_current": ("BIAS:CURR", self._bias_current_values)
+        }
+
+        if sweep_mode not in param_dict:
+            log.warning(
+                "Sweep mode but be one of %s, not '%s'.", list(param_dict.keys()), sweep_mode
+            )
+            return

```

Comment:
Why don't you raise an error?

Reply:
There's no reason, I've changed it to a KeyError now.